x-pack/solutions/observability/plugins/slo/server/services/slo_repository.ts

```diff
@@ -112,17 +115,48 @@ export class KibanaSavedObjectsSLORepository implements SLORepository {
   async search(
     search: string,
     pagination: Pagination,
-    options: { includeOutdatedOnly?: boolean } = { includeOutdatedOnly: false }
+    options: {
+      includeOutdatedOnly: boolean;
+      tags?: string;
+    } = { includeOutdatedOnly: false, tags: undefined }
   ): Promise<Paginated<SLODefinition>> {
+    const { includeOutdatedOnly, tags } = options;
+    let filter = null;
+    let tagString = '';
+    const splitTags = tags ? tags.split(',') : [];
+
+    if (splitTags.length > 1) {
+      tagString = '';
+      splitTags.forEach((tag: string, index: number) => {
+        if (tagString && tag) {
+          tagString += ` OR ${tag}`;
+        } else if (tag) {
+          tagString += `(${tag}`;
+        }
+
+        if (index === splitTags.length - 1) {
+          tagString += ')';
+        }
+      });
+    } else {
+      tagString = splitTags[0];
+    }
+
+    if (!!includeOutdatedOnly) {
+      filter = `slo.attributes.version < ${SLO_MODEL_VERSION}${
+        tags ? ` and slo.attributes.tags : ${tagString}` : ``
+      }`;
+    } else if (tags) {
+      filter = `slo.attributes.tags : ${tagString}`;
+    }
+
```

Comment:
Leverage `array.join()` instead of manipulating strings.

```suggestion
    const filters = [];
    if (tags.length > 0) {
      filters.push(`slo.attributes.tags: (${tags.join(" OR ")})`);
   }
   
    if (!!includeOutdatedOnly) {
      filters.push(`slo.attributes.version < ${SLO_MODEL_VERSION}`);
    }
    
    [...]
    
    ...(filters.length && { filter: filters.join(" AND ") }),
    
```